suite-native/blockchain/src/blockchainThunks.ts

```diff
@@ -0,0 +1,97 @@
+import { createThunk } from '@suite-common/redux-utils';
+import { NetworkSymbol, isNetworkSymbol } from '@suite-common/wallet-config';
+import {
+    blockchainActions,
+    fetchAndUpdateAccountThunk,
+    selectAccountByDescriptorAndNetworkSymbol,
+    selectAccountsByNetworkSymbol,
+    selectAccountsSymbols,
+    subscribeBlockchainThunk,
+} from '@suite-common/wallet-core';
+import { AccountKey } from '@suite-common/wallet-types';
+import { getNetwork } from '@suite-common/wallet-utils';
+import { BlockchainNotification } from '@trezor/connect';
+
+const actionsPrefix = '@native/blockchain';
+
+const accountLastFetchTime: Record<AccountKey, number> = {};
+const accountLastFetchTimeLimitMs = 1000 * 10; // 5 seconds
+
+const shouldRefetchAccount = ({
+    accountKey,
+    refetchLimitMs = accountLastFetchTimeLimitMs,
+}: {
+    accountKey: AccountKey;
+    refetchLimitMs?: number;
+}) => {
+    const lastFetchTime = accountLastFetchTime[accountKey];
+    if (!lastFetchTime) return true;
+    return Date.now() - lastFetchTime > refetchLimitMs;
+};
+
+export const syncAccountsWithBlockchainThunk = createThunk(
+    `${actionsPrefix}/syncAccountsThunk`,
+    async ({ symbol }: { symbol: NetworkSymbol }, { getState, dispatch }) => {
+        const accounts = selectAccountsByNetworkSymbol(getState(), symbol);
+        const accountForRefetch = accounts.filter(({ key }) =>
+            shouldRefetchAccount({ accountKey: key }),
+        );
+
+        const accountPromises = accountForRefetch.map(a =>
+            dispatch(fetchAndUpdateAccountThunk({ accountKey: a.key })),
+        );
+        accountForRefetch.forEach(a => {
+            accountLastFetchTime[a.key] = Date.now();
```

Comment:
this can be part of the map on line 40

Reply:
Using map for anything else than "mapping" to new value is bad idea. If you need to just trigger some use effect, forEach is good choice.